examples/src/main/scala/eu/neverblink/jelly/examples/PekkoStreamsDecoderFlow.scala

```diff
@@ -177,16 +188,27 @@ object PekkoStreamsDecoderFlow extends shared.ScalaExample:
   /**
    * Helper method to produce encoded data from a dataset.
    */
-  private def getEncodedData(dataset: Dataset)(using ActorSystem, ExecutionContext):
+  private def getEncodedData(dataset: Dataset)
+    (
+      using ActorSystem,
+      ExecutionContext,
+      JenaConverterFactory,
+      JenaAdapters.DATASET_ADAPTER.type,
+      JenaAdapters.MODEL_ADAPTER.type
+    ):
   (Seq[Array[Byte]], Seq[Array[Byte]], Seq[Array[Byte]]) =
-    val quadStream = RdfSource.builder.datasetAsQuads(dataset).source
-      .via(EncoderFlow.builder.withLimiter(ByteSizeLimiter(500)).flatQuads(JellyOptions.smallStrict).flow)
+    val quadStream = RdfSource.builder[Model, Dataset, Node, Triple, Quad]().datasetAsQuads(dataset).source
```

Comment:
Are you sure the explicit type parameters (here and below) are necessary? Scala should infer them from passed parameters (givens in this case).

Reply:
well, it doesnt